scripts/auto_reboot.sh

```diff
@@ -2,32 +2,25 @@
 # shellcheck source=/dev/null
 source "/home/steam/server/helper_functions.sh"
 
-if [ "${RCON_ENABLED,,}" != true ]; then
-    LogWarn "Unable to reboot. RCON is required."
-    exit 0
-fi
+if [ "${RCON_ENABLED,,}" = true ]; then
+    if [ "${AUTO_REBOOT_EVEN_IF_PLAYERS_ONLINE,,}" != true ]; then
+        players_count=$(get_player_count)
 
-if [ -z "${AUTO_REBOOT_WARN_MINUTES}" ]; then
-    LogError "Unable to auto reboot, AUTO_REBOOT_WARN_MINUTES is empty."
-    exit 0
-fi
+        if [ "$players_count" -gt 0 ]; then
+            LogWarn "There are ${players_count} players online. Skipping auto reboot."
+            exit 0
+        fi
+    fi
 
-if [ "${AUTO_REBOOT_EVEN_IF_PLAYERS_ONLINE,,}" != true ]; then
-  players_count=$(get_player_count)
-  if [ "$players_count" -gt 0 ]; then
-    LogWarn "There are ${players_count} players online. Skipping auto reboot."
-    exit 0
-  fi
-fi
-
-if [[ "${AUTO_REBOOT_WARN_MINUTES}" =~ ^[0-9]+$ ]]; then
-    for ((i = "${AUTO_REBOOT_WARN_MINUTES}" ; i > 0 ; i--)); do
-        RCON "broadcast The_Server_will_reboot_in_${i}_Minutes"
-        sleep "1m"
-    done
-    RCON save
-    RCON "shutdown 1"
-    exit 0
+    if countdown_message "${AUTO_REBOOT_WARN_MINUTES}" "The_Server_will_reboot"; then
+        shutdown_server
+    elif [ -z "${AUTO_REBOOT_WARN_MINUTES}" ]; then
+        LogError "Unable to auto reboot, the server is not empty and AUTO_REBOOT_WARN_MINUTES is empty"
+        exit 1
+    else
+        LogError "Unable to auto reboot, the server is not empty and AUTO_REBOOT_WARN_MINUTES is not an integer: ${AUTO_REBOOT_WARN_MINUTES}"
+    fi
```

Comment:
I think you can do these checks before you start the countdown. Would make it a bit cleaner in my opinion.

Reply:
I have it like that because if you so choose you could run the reboot script with no one in the server and it will reboot.

Also if it is a cron job and there is no one why not just reboot? The timer is supposed to just let players know and not prevent the reboot as that's what the enable flag is for

Reply:
Wouldn't it be better, if we added arg flags to the script instead of doing it interactively? Or both. Then we can call the script with the flags from the cronjob and if the user does not supply any flags it asks for them. That way we only have to manage one file.
Just an idea